src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,1388 @@
+     *     conjugated pi bond systems; bonds of higher order (>1) present in linear sidechains.
+     * </p>
+     * <p>
+     *     Extraction of marked substructures is done by creating deep copies of the cloned molecule's atoms and bonds,
+     *     and placing them in designated new IAtomContainer instances. During extraction, the preservation of the
+     *     chemical formula is checked, and a warning logged if not true.
+     *     The order of extraction is as follows: rings, conjugated pi bond systems and their fusion products;
+     *     additional double bonds connected to rings; isolated bonds of higher order;
+     *     atoms and bonds neighbouring tertiary or quaternary carbons; residual atoms and bonds as linear chains.
+     *     The extracted linear chains may also be fragmented according to the set maximum length
+     *     (see maxChainLengthSetting documentation).
+     *     Multiple settings may be activated for different algorithmic behavior regarding selected substructures
+     *     (see settings documentation).
+     *     After the extraction, the returned fragments are saturated with implicit hydrogen if the corresponding
+     *     setting is active.
+     * </p>
+     *
+     * @param aMolecule to fragment
+     * @return List of IAtomContainers containing the fragments (may be empty if no fragments are extracted)
+     * @throws NullPointerException if aMolecule is null
+     * @throws IllegalArgumentException if the given molecule cannot be fragmented
+     * @throws CloneNotSupportedException if cloning the given molecule is unsuccessful
+     */
+    @Override
+    public List<IAtomContainer> fragmentMolecule(IAtomContainer aMolecule)
+            throws NullPointerException, IllegalArgumentException, CloneNotSupportedException {
+        //skip fragmentation if molecule is not fragmentable and just return it as one fragment to keep it in pipeline
+        if ((boolean) aMolecule.getProperty(ASF_FILTER_MARKER) && this.keepNonFragmentableMoleculesSetting.get()) {
+            List<IAtomContainer> tmpNonFragACList = new ArrayList<>(1);
+            tmpNonFragACList.add(aMolecule);
+            return tmpNonFragACList;
+        }
+        //<editor-fold desc="Molecule Cloning and Chemical Formula Check" defaultstate="collapsed">
+        IAtomContainer tmpClone = aMolecule.clone();
+        this.chemObjectBuilderInstance = DefaultChemObjectBuilder.getInstance();
+        int tmpPreFragmentationAtomCount = 0;
+        for (IAtom tmpAtom: tmpClone.atoms()) {
+            if (tmpAtom.getAtomicNumber() != 0) {
+                tmpPreFragmentationAtomCount++;
+            }
+        }
+        LOGGER.log(Level.INFO, "PreFragAtomCount: " + tmpPreFragmentationAtomCount);
+        try {
+            AtomContainerManipulator.percieveAtomTypesAndConfigureAtoms(tmpClone);
+        } catch (CDKException aCDKException) {
+            LOGGER.log(Level.WARNING,
```

Comment:
In front of the static logger, "AlkylStructureFragmenter." indicates the class it belongs to. Analogous to the this-statement for instance variables.